source/RialtoGStreamerMSEAudioSink.cpp

```diff
@@ -345,40 +348,47 @@ static void rialto_mse_audio_sink_get_property(GObject *object, guint propId, GV
         return;
     }
 
-    std::shared_ptr<GStreamerMSEMediaPlayerClient> client = basePriv->m_mediaPlayerManager.getMediaPlayerClient();
+    const std::shared_ptr<GStreamerMSEMediaPlayerClient> &kClient = basePriv->m_mediaPlayerManager.getMediaPlayerClient();
```

Comment:
Missed this previously but I think we should revert back to std::shared_ptr<GStreamerMSEMediaPlayerClient> in these places, MediaPlayerClient can get released (releaseMediaPlayerClient) on a separate thread, taking a copy of the shared_ptr ensures we wait till finished with the object before it gets destructed.

Reply:
Well spotted!